### ISSUE TITLE
How can I find which media in the ConcatenatingMediaSource causes the load error?

### ISSUE THREAD

**Issue by chenjiancan** [google/ExoPlayer#6712]
I use ConcatenatingMediaSource to hold my play list, and I add a set of  ExtractorMediaSource items into it.

The media files with which ExtractorMediaSource create are all .mp4-s, but some files may brocken and they cause OnPlayerError callback with ExoPlaybackException.TYPE_SOURCE.

```
E/LoadTask: Unexpected exception handling load completed
    java.lang.NullPointerException
        at com.google.android.exoplayer2.util.Assertions.checkNotNull(Assertions.java:111)
        at com.google.android.exoplayer2.source.ExtractorMediaPeriod.onLoadCompleted(ExtractorMediaPeriod.java:504)
        at com.google.android.exoplayer2.source.ExtractorMediaPeriod.onLoadCompleted(ExtractorMediaPeriod.java:57)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:450)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.os.HandlerThread.run(HandlerThread.java:61)
 11-14 03:56:54.152 E/ExoPlayerImplInternal: Source error.
    com.google.android.exoplayer2.upstream.Loader$UnexpectedLoaderException: Unexpected NullPointerException: null
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:454)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.os.HandlerThread.run(HandlerThread.java:61)
     Caused by: java.lang.NullPointerException
        at com.google.android.exoplayer2.util.Assertions.checkNotNull(Assertions.java:111)
        at com.google.android.exoplayer2.source.ExtractorMediaPeriod.onLoadCompleted(ExtractorMediaPeriod.java:504)
        at com.google.android.exoplayer2.source.ExtractorMediaPeriod.onLoadCompleted(ExtractorMediaPeriod.java:57)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:450)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.os.HandlerThread.run(HandlerThread.java:61) 
```

My question is how could I log the file name(or uri) and remove it from the ConcatenatingMediaSource, and then coutinue play?

I'd try ConcatenatingMediaSource.removeMediaSource(player.currentWindowIndex), but 
1. player.currentWindowIndex is not the item cause the error, maybe "nextWindowIndex", but
2. how to find the file,  notice that shuffleModeEnabled is true

**Comment by chenjiancan** [comment]
I also found that ExtractorMediaSource.factory has a tag field, should I set the file uri to "tag", and get it when needed.

**Comment by tonihei** [comment]
`player.currentWindowIndex` should be the index of the media source that caused the error. 

There was a bug previously (#5407) where the index wasn't reported correctly, but this has been fixed in 2.11.0. So I assumes this is a duplicate of this issue.
